src/Network/Transport/TCP.hs

```diff
@@ -854,7 +866,9 @@ handleConnectionRequest transport sock = handle handleException $ do
     forM_ (tcpUserTimeout $ transportParams transport) $
       N.setSocketOption sock N.UserTimeout
     ourEndPointId <- recvWord32 sock
-    theirAddress  <- EndPointAddress . BS.concat <$> recvWithLength sock
+    let maxAddressLength = tcpMaxAddressLength $ transportParams transport
+    theirAddress  <- EndPointAddress . BS.concat <$>
+      recvWithLength maxAddressLength sock
```

Comment:
This throws an exception, which will cause `terminationHandler` to run, which will close the transport. Shouldn't it be handled differently?